### ISSUE TITLE
Markdown is ignored in same line with doxygen command

### ISSUE THREAD

**Issue by ricric86** [doxygen/doxygen#7999]
**Describe the bug**
Markdown is not resolved when using it in the same line as doxygen commands. For example after commands of \brief or \details markdown commands such as "### Heading" are ignored.

**To Reproduce**
The following line
`/// @details ### Check Test`

will produce an output like this
`### Check Test`

**Expected behavior**
Markdown would be resolved as for example a heading. This was possible in older versions of doxygen.
In the example the line
`/// @details ### Check Test`
would produce a text in a real heading like
> ### Check Test

**Version**
* `Doxygen 1.8.20 (f246dd2f1c58eea39ea3f50c108019e4d4137bd5)` 
* running on Windows 10

**Comment by albert-github** [comment]
My thoughts:

What is the objection against writing
```
/// @details
/// ### Check Test
```

In a brief description a heading should nt be present, a brief description is just for a brief description.

**Comment by ricric86** [comment]
Yes, I will not need a heading for `@brief` and I could cope with writing the markdown heading in the next line as you proposed.

I found the hint that using a markdown heading requires to be at the start of the line in [Markdown Support](https://www.doxygen.nl/manual/markdown.html#md_headers)
Hmm but why was this possible in the past?

It seems it's just a problem with `@details` and the heading. Other markdown commands are working within the same line. The general statement I made that markdown is ignored is wrong.
